core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java

```diff
@@ -262,6 +266,51 @@ public ServerListenerBuilder whenStopped(Iterable<? extends Consumer<? super Ser
         return this;
     }
 
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It will wait indefinitely for the {@link ExecutorService} to terminate during shutdown.
+     */
+    @UnstableApi
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService) {
+        requireNonNull(executorService, "executorService");
+        serverStoppingCallbacks.add(s -> ShutdownSupport.of(executorService).shutdown());
+        return this;
+    }
+
+    /**
+     * Add a callback that gracefully shuts down the given {@link ExecutorService} when the {@link Server}
+     * is stopping. It allows a maximum duration of {@code terminationTimeout} for the {@link ExecutorService}
+     * to terminate gracefully before it is forcefully terminated.
+     */
+    @UnstableApi
+    public ServerListenerBuilder shutdownWhenStopping(ExecutorService executorService,
+                                                      Duration terminationTimeout) {
```

Comment:
What do you think of:
- `terminationTimeout == 0`: wait indefinitely
- `terminationTimeout > 0`: wait for the specified time
- `terminationTimeout < 0`: throw an exception

Reply:
Thank you for the suggestion! I have removed the `shutdownWhenStopping(ExecutorService executorService)`. I have also added comments explaining `terminationTimeout`.